### PR TITLE
[ie/YoutubeSearchDate] Fix search parameters

### LINKED ISSUE #9156: [Youtube search] Certain queries do not yield results
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting that yt-dlp is broken on a **supported** site
- [X] I've verified that I have **updated yt-dlp to nightly or master** ([update instructions](https://github.com/yt-dlp/yt-dlp#update-channels))
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [ ] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

Spain

### Provide a description that is worded well enough to be understood

When searching in Youtube, certain search terms might trigger an anti-harm message. This causes yt-dlp to not download/extract anything.

Command used (it's a Nirvana song):
`yt-dlp "ytsearch1:i hate myself and i wanna die"`

yt-dlp does not report an error, but it does mention 0 items downloaded. Output:

```
[youtube:search] Extracting URL: ytsearch1:i hate myself and i wanna die
[download] Downloading playlist: i hate myself and i wanna die
[youtube:search] query "i hate myself and i wanna die": Downloading web client config
[youtube:search] query "i hate myself and i wanna die" page 1: Downloading API JSON
[youtube:search] Playlist i hate myself and i wanna die: Downloading 0 items
[download] Finished downloading playlist: i hate myself and i wanna die
```

When searching in a browser, it shows the anti-harm message without any results, but there is a button that says "show results anyway", which (in this particular case) adds the URL query parameter `sp=8AEB`. I have not investigated what this does or what the value means, but the same value seems to work regardless of what the search query is. Searching in an incognito window with the parameter added from the beggining seems to bypass the message and show the results directly.

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [ ] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
[debug] Command-line config: ['-vU']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2023.12.30 from yt-dlp/yt-dlp [f10589e34]
[debug] Lazy loading extractors is disabled
[debug] Python 3.11.6 (CPython x86_64 64bit) - Linux-6.7.3-zen1-1-zen-x86_64-with-glibc2.38 (OpenSSL 3.2.1 30 Jan 2024, glibc 2.38)
[debug] exe versions: ffmpeg 6.1.1 (setts), ffprobe 6.1.1
[debug] Optional libraries: Cryptodome-3.12.0, certifi-2023.11.17, requests-2.31.0, sqlite3-3.45.1, urllib3-1.26.18
[debug] Proxy map: {}
[debug] Request Handlers: urllib, requests
[debug] Loaded 1832 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: stable@2023.12.30 from yt-dlp/yt-dlp
yt-dlp is up to date (stable@2023.12.30 from yt-dlp/yt-dlp)
```

### PR DESCRIPTION
**Pull request by alb**
### Fix search parameters for YouTube extractor 

This PR fixes the search parameters for `YouTubeSearchDateIE` to use the following filters: `sort:date`, `filter:videos`.
I didn't update the parameters for `YouTubeSearchIE` as the current parameters still seem to be correct and testing on the website gives me the same parameters when applying the`filter:videos` filter manually.

Fixes #9156 

<details open>
<summary>Template</summary>

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details>

### CODE REVIEW

**Comment by pukkandan** [comment]
SearchIE has the same issue. Pls update it too. I've mentioned the desired param in #9156. Also, add a test - perhaps the example in the issue

**Comment by alb** [comment]
@pukkandan I've updated the search params for SearchIE as well now and added the example from the issue as a test along with a note as to why it's needed.

File: yt_dlp/extractor/youtube.py [hunk 1]
```diff
@@ -6965,22 +6965,29 @@ class YoutubeSearchIE(YoutubeTabBaseInfoExtractor, SearchInfoExtractor):
...
         'url': 'ytsearch5:youtube-dl test video',
         'playlist_count': 5,
         'info_dict': {
             'id': 'youtube-dl test video',
             'title': 'youtube-dl test video',
         }
+    }, {
+        'note': 'Suicide/self-harm search warning',
+        'url': 'ytsearch1:i hate myself and i wanna die',
+        'info_dict': {
```
Comment (bashonly) [inline yt_dlp/extractor/youtube.py:6979]:
```suggestion
        'url': 'ytsearch1:i hate myself and i wanna die',
        'playlist_count': 1,
        'info_dict': {
```

**Review by bashonly — APPROVED** [review]

**Comment by bashonly** [comment]
LGTM. In the future, please don't force push, all PR branch commits with be squashed upon merge

File: yt_dlp/extractor/youtube.py [hunk 2]
```diff
@@ -6965,22 +6965,29 @@ class YoutubeSearchIE(YoutubeTabBaseInfoExtractor, SearchInfoExtractor):
     IE_DESC = 'YouTube search'
     IE_NAME = 'youtube:search'
     _SEARCH_KEY = 'ytsearch'
-    _SEARCH_PARAMS = 'EgIQAQ%3D%3D'  # Videos only
+    _SEARCH_PARAMS = 'EgIIAfABAQ=='  # Videos only
```
Comment (pukkandan) [inline yt_dlp/extractor/youtube.py:6968]: Are you sure this is correct? The previous params were % encoded

Comment (bashonly) [reply yt_dlp/extractor/youtube.py:6968]: Tested and works. They are encoded in the query anyways

File: yt_dlp/extractor/youtube.py [hunk 3]
```diff
@@ -6965,22 +6965,30 @@ class YoutubeSearchIE(YoutubeTabBaseInfoExtractor, SearchInfoExtractor):
...
     _TESTS = [{
         'url': 'ytsearch5:youtube-dl test video',
         'playlist_count': 5,
```
Comment (bashonly) [inline yt_dlp/extractor/youtube.py:6971]: the new params yield only 2 results, while the old params still yield 5

**Review by bashonly — CHANGES_REQUESTED** [review]

File: yt_dlp/extractor/youtube.py [hunk 4]
```diff
@@ -6965,22 +6965,30 @@ class YoutubeSearchIE(YoutubeTabBaseInfoExtractor, SearchInfoExtractor):
     IE_DESC = 'YouTube search'
     IE_NAME = 'youtube:search'
     _SEARCH_KEY = 'ytsearch'
-    _SEARCH_PARAMS = 'EgIQAQ%3D%3D'  # Videos only
+    _SEARCH_PARAMS = 'EgIIAfABAQ=='  # Videos only
```
Comment (pukkandan) [inline yt_dlp/extractor/youtube.py:6968]:
```suggestion
    _SEARCH_PARAMS = 'EgIQAfABAQ=='  # Videos only
```

File: yt_dlp/extractor/youtube.py (hunk 3 above)
Comment (pukkandan) [reply yt_dlp/extractor/youtube.py:6971]: my bad. Seems I copied the wrong params. What we want is https://github.com/iv-org/invidious/blame/99a5e9cbc44daa4555f36b43bc4b7246aee454c9/spec/invidious/search/yt_filters_spec.cr#L23

**Review by bashonly — APPROVED** [review]

**Pull request merged**